### PR TITLE
Easier integration into build system

### PR DESCRIPTION
**Pull request by schulz0r**
Hello,

I use CT as a backend for my helicopter path planner and struggled to integrate it into my existing build system. I encountered two problems with the existing install script: First of all, it automatically installs each module after building it so that interdependencies of consecutive modules can be resolved. This needs privileged access which is an annoyance when building my project. Second, without a top level CMakeLists.txt, you cannot simply add CT as an external project to your own project.
Since profound changes to this project are most likely rejected, I did the bare minimum to make CT work with my code. I added a root CMakeLists.txt so that it can be easily integrated as an external project. Since all modules are on the same hierarchy level, but have interdependencies, I had to cache the targets like 
`set(ct_core ct_core CACHE INTERNAL "ct_core library target.")`
This causes `ct_core` to be found by the other modules without the need of installing every single module.
Basically, that's it, but I encountered another problem when building CT on a linux machine without Python 3 libs installed. In ct_core/CMakeLists.txt, CMake tried to include Python headers even thou they have not been found, which is now being prevented. Your build script should still work fine btw. 
This is my second try on this problem and I hope that this solution is now acceptable for you.

Regards

Schulz0r

### CODE REVIEW

**Comment by schulz0r** [comment]
Please don't merge right now! I think I figured out a more minimalistic way which does not even need a top level CMakeLists.txt. I can add single modules to my build system like it is originally intended instead. Still I need some minor modifications.
I will report back tomorrow.

**Comment by schulz0r** [comment]
Good news, it's working without a CMakeLists.txt in the root directory of this project. With the latest update, I can use CT as a backend for my path planner software. Please review my changes to your project and accept if they are okay.

**Comment by markusgft** [comment]
I tried it in combination with `catkin tools`, it works flawlessly. Thanks for contributing this!
What is the trick that you are now employing to include it in your project?

**Review by markusgft — APPROVED** [review]

**Comment by schulz0r** [comment]
First of all, I use CMake to include external projects which are not available in software repositories. The trick is described [here](https://github.com/google/googletest/blob/master/googletest/README.md).
I have a CMakeLists-ControlToolbox.txt.in which contains the externalproject_add() function call. Then, I can automatically add CT in case it is not installed on the system like this:
```
if(NOT ${ControlToolbox_FOUND})
    # Download and unpack CT at configure time

	configure_file(CMakeLists-ControlToolbox.txt.in ControlToolbox-download/CMakeLists.txt)
	execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" -DCMAKE_PREFIX_PATH=${BINARY_DIR} .
	  RESULT_VARIABLE result
	  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ControlToolbox-download/ )

	if(result)
	  message(FATAL_ERROR "CMake step for ControlToolbox failed: ${result}")
	endif()

        # clone project during configure time to src directory

	execute_process(COMMAND ${CMAKE_COMMAND} --build .
	  RESULT_VARIABLE result
	  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ControlToolbox-download/ )

	if(result)
	  message(FATAL_ERROR "Build step for ControlToolbox in commandmodel failed: ${result}")
	endif()

        # now add all CT needed modules

	add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/ControlToolbox-src/ct_core
	  ${CMAKE_CURRENT_BINARY_DIR}/ControlToolbox-build/ct_core
	  EXCLUDE_FROM_ALL)
   	add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/ControlToolbox-src/ct_optcon
	     	${CMAKE_CURRENT_BINARY_DIR}/ControlToolbox-build/ct_optcon
      		EXCLUDE_FROM_ALL)
endif()
```

**Pull request merged**
